core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -4721,24 +4551,132 @@ public DisplayType getDisplayType() {
     return mDisplayType;
   }
 
+  private static final Map<PropertyKey, Deprecated> DEPRECATED_KEYS =
+      populateAnnotatedKeyMap(PropertyKey.class, PropertyKey.class, Deprecated.class);
+  private static final Map<Template, Deprecated> DEPRECATED_TEMPLATES =
+      populateAnnotatedKeyMap(Template.class, Template.class, Deprecated.class);
+
+  private static final Map<PropertyKey, Removed> REMOVED_KEYS =
+      populateAnnotatedKeyMap(RemovedKey.class, PropertyKey.class, Removed.class);
+  private static final Map<Template, Removed> REMOVED_TEMPLATES =
+      populateAnnotatedKeyMap(Template.class, Template.class, Removed.class);
+
   /**
-   * @param name the name of a property key
+   * Given a class to search, a field type, and an annotation type will return a map of all
+   * fields which are marked with the given annotation to the instance of the annotation.
+   *
+   * @param searchClass the class to search through for fields
+   * @param fieldType the class of the field to search for
+   * @param annotationClazz the annotation to look for
+   * @param <I> any class
+   * @param <J> any class
+   * @param <K> a class extending Annotation
+   * @return a map
+   */
+  private static <I, J, K extends Annotation> Map<J, K> populateAnnotatedKeyMap(
+      Class<I> searchClass, Class<J> fieldType, Class<K> annotationClazz) {
+    Map<J, K> annotations = new HashMap<>();
+    for (Field field : searchClass.getDeclaredFields()) {
+      if (!field.getType().equals(fieldType)) {
+        continue;
+      }
+
+      K keyAnnotation = field.getAnnotation(annotationClazz);
+
+      try {
+        // Field#get parameter can be null if retrieving a static field (all PKs are static)
+        // See https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Field.html
+        // This also works with Template enums
+        J key = fieldType.cast(field.get(null));
+        if (keyAnnotation != null) {
+          annotations.put(key, keyAnnotation);
+        }
+      } catch (IllegalAccessException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return annotations;
+  }
+
+  private static <J extends Annotation> J getKeyAnnotation(Map<PropertyKey, J> keyAnnotationMap,
+      Map<Template, J> templateAnnotationMap, PropertyKey key) {
+    if (keyAnnotationMap.containsKey(key)) {
+      return keyAnnotationMap.get(key);
+    } else {
+      for (Map.Entry<Template, J> e : templateAnnotationMap.entrySet()) {
+        Matcher match = e.getKey().match(key.getName());
+        if (match.matches()) {
+          return e.getValue();
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns whether or not the given property key is deprecated or completely removed.
```

Comment:
I can't tell from this sentence whether the method returns `true` vs `false` for completely removed keys

Reply:
I didn't properly update this javadoc when I refactored. Will fix